library/X509/CertificateUtils.php

```diff
@@ -365,6 +367,62 @@ private static function findOrInsertDn($db, $certInfo, $type)
         return $hash;
     }
 
+    /**
+     * Cleanup certificate chains that aren't used by any targets
+     *
+     * @param Connection $conn
+     */
+    public static function cleanupCertificateChains(Connection $conn)
```

Comment:
I would combine all three methods into one because call order is important and using any of them alone doesn't make sense: `cleanupCertificatesNoLongerUsed()`.